src/deck/drivers/src/usddeck.c

```diff
@@ -730,32 +740,29 @@ static void usdWriteTask(void* prm)
         uint16_t version = 1;
         usdWriteData(&version, sizeof(version));
 
-        uint16_t numEventTypes = usdLogConfig.numEventtriggers;
-        if (usdLogConfig.hasFixedFrequencyEvent) {
-          ++numEventTypes;
-        }
+        uint16_t numEventTypes = usdLogConfig.numEventConfigs;
         usdWriteData(&numEventTypes, sizeof(numEventTypes));
 
-        for (uint16_t id = 0; ; ++id) {
-          eventtrigger* et = eventtriggerGetById(id);
-          if (!et) {
-            break;
-          }
-          DEBUG_PRINT("eteM %d\n", et->enableMask);
-          // if (et->enableMask & (1<<eventtriggerHandler_USD)) {
-          if (true) {
-            // write header for each enabled event
-            usdWriteData(&id, sizeof(id));
-            usdWriteData(et->name, strlen(et->name) + 1);
-            uint16_t numVariables = et->numPayloadVariables;
-            usdWriteData(&numVariables, sizeof(numVariables));
+        for (int i = 0; i < numEventTypes; ++i) {
+          usdLogEventConfig_t* cfg = &usdLogConfig.eventConfigs[i];
+          const eventtrigger *et = eventtriggerGetById(cfg->eventId);
+          uint16_t numVariables = cfg->numVars;
 
-            for (int i = 0; i < numVariables; ++i)
-            {
-              usdWriteData(et->payloadDesc[i].name, strlen(et->payloadDesc[i].name));
+          usdWriteData(&cfg->eventId, sizeof(cfg->eventId));
+
+          if (cfg->eventId == FIXED_FREQUENCY_EVENT_ID) {
+            usdWriteData(FIXED_FREQUENCY_EVENT_NAME, strlen(FIXED_FREQUENCY_EVENT_NAME) + 1);
```

Comment:
Is it documented somewhere that strings written should include the termination byte?

Reply:
This is now documented in micro-sd-card-deck.md, as part of the binary file format description.